src/utils/dateInfo.js

```diff
@@ -396,6 +397,8 @@ export default class DateInfo {
   }
 
   intersectsDay(day) {
+    // console.count('dateInfo intersectsDay');
+    // return null;
```

Comment:
Returning null here shows that this is an expensive comparison. No suprises here. there may be room for optimisation here, but the optimisations will only be linear.